content/blog/2022-03-22-hot-swapping-python-lambda-functions-using-localstack/index.md

```diff
@@ -0,0 +1,227 @@
+
+Let's get started with LocalStack. To install LocalStack, you need to ensure that the LocalStack CLI is installed. Through `pip`, you can easily do that using the following command:
+
+```sh
+pip install localstack
+```
+
+It will install the `localstack-cli` which is used to run the Docker image that hosts the LocalStack runtime. You can start LocalStack in a detached mode by running the following command:
+
+```sh
+localstack start -d
+```
+
+For the purpose of this blog, we will use a sample AWS Lambda function from the [AWS SDK examples](https://github.com/awsdocs/aws-doc-sdk-examples/blob/main/python/example_code/lambda/lambda_handler_basic.py). Here is our Lambda function:
+
+
+```python
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+"""
+Purpose
+
+Shows how to implement an AWS Lambda function that handles input from direct
+invocation.
+"""
+
+# snippet-start:[python.example_code.lambda.handler.calculate]
+import logging
+import math
+
+logger = logging.getLogger()
+logger.setLevel(logging.INFO)
+
+# Define a list of Python lambda functions that are called by this AWS Lambda function.
+ACTIONS = {
+    'square': lambda x: x * x,
+    'square root': lambda x: math.sqrt(x),
+    'increment': lambda x: x + 1,
+    'decrement': lambda x: x - 1,
+}
+
+
+def lambda_handler(event, context):
+    """
+    Accepts an action and a number, performs the specified action on the number,
+    and returns the result.
+
+    :param event: The event dict that contains the parameters sent when the function
+                  is invoked.
+    :param context: The context in which the function is called.
+    :return: The result of the specified action.
+    """
+    logger.info('Event: %s', event)
+
+    result = ACTIONS[event['action']](event['number'])
+    logger.info('Calculated result of %s', result)
+
+    response = {'result': result}
+    return response
+# snippet-end:[python.example_code.lambda.handler.calculate]
+```
+
+To get started with hot swapping the above Lambda function, start the LocalStack container by configuring the `LAMBDA_REMOTE_DOCKER`:
+
+```sh
+LAMBDA_REMOTE_DOCKER=0 localstack start -d --host
+```
+
+We are running LocalStack in host mode, which might install more developer dependencies and is not supported on Windows. The `LAMBDA_REMOTE_DOCKER` is configured as `false` to ensure that the Lambda volume mounts works while we are mounting a temporary folder on the host. If you are using a Docker-Compose setup to start LocalStack, you can add the following option to your `docker-compose.yml` file:
+
```

Comment:
I think we should not rely on host mode for the tutorial. It's not something we support for normal use. Can we maybe try to make it work without it and re-write this section accordingly?